lib/ansible/module_utils/urls.py

```diff
@@ -954,7 +954,7 @@ def open(self, method, url, data=None, headers=None, use_proxy=None,
                 passman = urllib_request.HTTPPasswordMgrWithDefaultRealm()
 
                 # this creates a password manager
-                passman.add_password(None, netloc, username, password)
+                passman.add_password(auth_realm, netloc, username, password)
```

Comment:
Unit tests in `test/units/module_utils/urls/test_Request.py` should be extended to validate that this value is being inserted as expected.  Likely via 2 new test cases to validate with and without a value.  `test_Request_open_username` should give you enough to go on for how to achieve this.